### PR TITLE
Add instructions on how you can debug Breeze easily

### PR DESCRIPTION
**Pull request by potiuk**
It could be unclear how to debug Breeze using IDE debuggers, this change provides an explanation including screenshot from PyCharm.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Since we are going to work on more of changes similar to https://github.com/apache/airflow/pull/35086 I've figured that it might be non-obvious how to easily configure your IDE to run debugging sessions - without involving remote debugger or using ipython or "print debugging".

It's rather easy - but it's not something you'd usually do when you develop Pythin script. Breeze is an installable  package which has automated upgrade checks and uses click to pass commands, so you need to use `module` and add `SKIP_UPGRADE_CHECK` variable in "IDE "debugger".

I added screenshot and explanation about it in Breeze docs. Might be also helpful to pull in others who would like to

**Review by eladkal — APPROVED** [review]
Nice!

File: BREEZE.rst [hunk 1]
```diff
@@ -2649,3 +2649,50 @@ This will also remove breeze from the folder: ``${HOME}.local/bin/``
...
+For testing, you can create your own virtual environment, or use the one that ``pipx`` created for you if you
+already installed breeze following the recommended ``pipx install -e ./dev/breeze`` command.
+
+For local virtualenv, you can use ``pyenv`` or any other virtualenv wrapper. For example with ``pyenv``,
+you can use ``pyenv virtualenv 3.8.6 airflow-breeze`` to create virtualenv called ``airflow-breeze``
+with Python 3.8.6. Then you can use ``pyenv activate airflow-breeze`` to activate it and install breeze
+in editable mode with ``pip install -e ./dev/breeze``.
+
+For ``pipx`` virtualenv, you can use the virtualenv that ``pipx`` created for you. You can find the name
+where ```pipx`` keeps their venvs via ``pipx list`` command. Usually it is
```
Comment (hussein-awala) [inline BREEZE.rst]:
```suggestion
where ``pipx`` keeps their venvs via ``pipx list`` command. Usually it is
```

**Review by hussein-awala — APPROVED** [review]

File: BREEZE.rst [hunk 2]
```diff
@@ -2649,3 +2649,50 @@ This will also remove breeze from the folder: ``${HOME}.local/bin/``
...
+with Python 3.8.6. Then you can use ``pyenv activate airflow-breeze`` to activate it and install breeze
+in editable mode with ``pip install -e ./dev/breeze``.
+
+For ``pipx`` virtualenv, you can use the virtualenv that ``pipx`` created for you. You can find the name
+where ``pipx`` keeps their venvs via ``pipx list`` command. Usually it is
+``${HOME}/.local/pipx/venvs/apache-airflow-breeze`` where ``$HOME`` is your home directory.
+
+The venv can be used for running breeze tests and for debugging breeze. While running tests should
+be usually "out-of-the-box" for most IDEs, once you configure ``./dev/breeze`` project to use the venv,
+Running/debugging a particular breeze command you want to debug might be a bit more tricky.
```
Comment (amoghrajesh) [inline BREEZE.rst]: Super nit: running/debugging

You can ignore as well

File: BREEZE.rst [hunk 3]
```diff
@@ -2649,3 +2649,50 @@ This will also remove breeze from the folder: ``${HOME}.local/bin/``
...
+
+When you configure your "Run/Debug configuration" to run breeze command you should
+make sure of those things:
```
Comment (amoghrajesh) [inline BREEZE.rst]: Nit: ensure truth of the points below

**Review by amoghrajesh — APPROVED** [review]
Nice! Some nits which you can ignore as well

**Pull request merged**